src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java

```diff
@@ -237,6 +240,32 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
       return this;
     }
 
+  /**
+   * For a given server endpoint, validate according to RFC 2396
+   *
+   * @return {@code this}
+   * @throws IllegalStateException
+   */
+    public Builder validateEndpoint() throws IllegalStateException {
+      URL url = null;
+
+      try {
+        url = new URL(this.server);
+      } catch (MalformedURLException e) {
+        throw new IllegalStateException(e);
```

Comment:
pass in detail msg along with the exception
throw new IllegalStateException("Please provide valid Wavefront endpoint", e);